packages/material-ui/src/Select/SelectInput.test.js

```diff
@@ -1,178 +1,242 @@
+
+      fireEvent.click(getByRole('button'));
+      act(() => {
+        clock.tick(99);
+      });
+
+      expect(onEntered.callCount).to.equal(0);
+
+      act(() => {
+        clock.tick(1);
+      });
+
+      expect(onEntered.callCount).to.equal(1);
     });
 
     it('should be able to override PaperProps minWidth', () => {
-      const wrapper = shallow(
-        <SelectInput {...defaultProps} MenuProps={{ PaperProps: { style: { minWidth: 12 } } }} />,
+      const { getByTestId } = render(
+        <SelectInput
+          classes={{}}
+          IconComponent="div"
+          MenuProps={{ PaperProps: { 'data-testid': 'paper', style: { minWidth: 12 } } }}
+          open
+          value="10"
+        >
+          <MenuItem value="10">Ten</MenuItem>
+        </SelectInput>,
       );
-      assert.strictEqual(wrapper.find(Menu).props().PaperProps.style.minWidth, 12);
+
+      expect(getByTestId('paper').style).to.have.property('minWidth', '12px');
     });
   });
 
   describe('prop: SelectDisplayProps', () => {
-    it('should apply additional props to the clickable div element', () => {
-      const wrapper = shallow(
-        <SelectInput {...defaultProps} SelectDisplayProps={{ 'data-test': 'SelectDisplay' }} />,
+    it('should apply additional props to trigger element', () => {
+      const { getByRole } = render(
+        <SelectInput
+          classes={{}}
+          IconComponent="div"
+          SelectDisplayProps={{ 'data-test': 'SelectDisplay' }}
+          value="10"
+        >
+          <MenuItem value="10">Ten</MenuItem>
+        </SelectInput>,
       );
 
-      const selectDisplay = wrapper.find('[role="button"]');
-      assert.strictEqual(selectDisplay.props()['data-test'], 'SelectDisplay');
+      expect(getByRole('button')).to.have.attribute('data-test', 'SelectDisplay');
     });
   });
 
   describe('prop: type', () => {
     it('should be hidden by default', () => {
-      const wrapper = shallow(<SelectInput {...defaultProps} />);
-      assert.strictEqual(wrapper.find('input').props().type, 'hidden');
-    });
+      const { container } = render(
+        <SelectInput classes={{}} IconComponent="div" value="10">
+          <MenuItem value="10">Ten</MenuItem>
+        </SelectInput>,
+      );
 
-    it('should be able to override it', () => {
-      const wrapper = shallow(<SelectInput {...defaultProps} type="text" />);
-      assert.strictEqual(wrapper.find('input').props().type, 'text');
+      expect(container.querySelector('input')).to.have.property('type', 'hidden');
     });
-  });
 
-  it('should render a placeholder if there is no value to display', () => {
-    const { getByRole } = render(<SelectInput {...defaultProps} renderValue={() => '   '} />);
-    expect(getByRole('button').querySelector('span')).to.be.ok;
+    /**
+     * TODO how do we support this? In its current state it will log a warning
+     * with no way to fix it
```

Comment:
I think we should just kill the `native` prop. This makes it incredible hard to reason about allowed props. I don't think anybody is actually changing this prop dynamically at which point requiring either `Select` or `NativeSelect` should be preferred. 

At that point we can come up with a better name for this based on usage. I guess our input components use it to move focus to the element which can toggle the open state.

Reply:
The only way to trigger this is by using `<Select inputProps={{ type: 'text' }}` />` which is useless both for `native={true}` and `native={false}`. We explicitly remove it by default https://github.com/mui-org/material-ui/blob/fdc7de2d5ec95873a48f37937c1b3bb35d864773/packages/material-ui/src/Select/Select.js#L67

I guess this can be safely removed in v5.

Reply:
> The only way to trigger this is by using `<Select inputProps={{ type: 'text' }} />`

I think that this case should be ignored.

Reply:
Alright let's remove it.